DANS-KNAW/easy-solr4files-index
# NO JIRA: Solid solutions for quick fixes

Fixes EASY-

#### When applied it will
* Make filter on depositor ID (#14) maintainable
  (having the fields to skip too far apart from the desired fields
  might cause trouble when having to maintain)
* Normalize white space (#15) only for complex xml elements in the DDM (in practice the creator)

#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?

Preparations:
* Build and provision the local VM, make sure you have your local snapshot installed:
  `yum list installed dans*` this might fail if the blessed branch incremented the version number after the branch under test was created
* Create test bag(s) in the VM.
  For example when having a bag "`data-`XXX" in the root of the project, do in the vm:
  `mkdir -p /data/bag-stores/pdbs/00/000000000000000000000000000001/`
  `cp -r /vagrant/data-`XXX` /data/bag-stores/pdbs/00/000000000000000000000000000001/`
  Make sure there are no hidden files (`.DS_Store`) in the bags that should not be there:
  `find /data/bag-stores -name ".*"  -exec rm "{}" \;`

Actions:
* populate the index with `easy-solr4files-index init`
* `curl -i 'http://user001:user001@test.dans.knaw.nl:20150/filesearch?text=`YYY`'`
  or in your browser: `http://test.dans.knaw.nl:8983/solr/#/fileitems/query` with `easy_*` in `fl`

Expected results:
* The curl command should not return a `easy_dataset_depositor_id` with `EASY-User-Account` from the `bag-info.txt`, the web-ui will.
* The file-path and other fields should be a verbatim copy of the fields in dataset.xml and files.xml, each creator should be a one-liner with just single spaces.

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..
